woocommerce/woocommerce-gateway-stripe
# Refactor post meta calls to make compatible with HPOS

Fixes #2538 

## Changes proposed in this Pull Request:
- Replaced all the `*_post_meta*` calls with HPOS compatible methods.
- Replaced `get_post_meta` with `$order->get_meta`
- Replaced `delete_post_meta` with `$order->delete_meta_data`

## Testing instructions
- Make sure there are no remaining `*_post_meta*` calls in the code that has not been converted to HPOS compatible method.
- Test with both HPOS enabled and disabled states. Go to **WooCommerce -> Advanced -> Features** and select the desired option from `Data storage for orders` to enable or disable the feature.

### Test subscription renewal 
- Purchase a subscription as a shopper (`user.a@test.com`) 
- From **WooCommerce -> Status -> Scheduled actions** find the `woocommerce_scheduled_subscription_payment` action for this subscription id and run the action. The subscription should renew successfully without any error in the logs.
- Manually renew the subscription a the shopper from the `My Account -> Subscriptions` tab. The subscription should renew successfully without any error in the logs.
- Cancel the subscription from **WooCommerce -> Subscriptions**. Now as a shopper resubscribe to this subscription from 
`My Account -> Subscriptions` tab. A new subscription should be created without any errors. 

### Test export
- As the previous user (`user.a@test.com`) place a couple of non-subscription orders and an order with `COD`.
- In `wp-admin` go to **Tools -> Export Personal Data**. Use `user.a@test.com` in the email field of the data export request section and send request.
- From the table below click `Download personal data` (below the email). 
- From the downloaded zip open the HTML file. Make sure the page includes the orders and the subscriptions and there is no errors in the log.
